tools/importer-rest-api-specs/components/testing/block_value_for_field.go

```diff
@@ -18,27 +18,30 @@ func (tb TestBuilder) getBlockValueForField(field resourcemanager.TerraformSchem
 	}
 
 	if field.ObjectDefinition.Type == resourcemanager.TerraformSchemaFieldTypeList || field.ObjectDefinition.Type == resourcemanager.TerraformSchemaFieldTypeSet {
-		// TODO: support for Lists of Basic Types
-		if field.ObjectDefinition.NestedObject.Type != resourcemanager.TerraformSchemaFieldTypeReference {
-			return nil, fmt.Errorf("internal-error: Lists and Sets currently only support a Reference - but got %q", string(field.ObjectDefinition.NestedObject.Type))
-		}
+		var nestedBlock *hclwrite.Block
+		var err error
 
-		nestedModelName := *field.ObjectDefinition.NestedObject.ReferenceName
-		nestedModel, ok := tb.details.SchemaModels[nestedModelName]
-		if !ok {
-			return nil, fmt.Errorf("the schema model %q referenced by nested List/Set %q was not found", nestedModelName, field.HclName)
+		if field.ObjectDefinition.NestedObject.Type == resourcemanager.TerraformSchemaFieldTypeReference {
+			nestedModelName := *field.ObjectDefinition.NestedObject.ReferenceName
+			nestedModel, ok := tb.details.SchemaModels[nestedModelName]
+			if !ok {
+				return nil, fmt.Errorf("the schema model %q referenced by nested List/Set %q was not found", nestedModelName, field.HclName)
+			}
+
+			// first go pull out the nested item
+			nestedBlock, err = tb.getBlockValueForModel(field.HclName, nestedModel, dependencies, onlyRequiredFields)
+			if err != nil {
+				return nil, fmt.Errorf("retrieving block value for field %q containing nested List/Set reference to %q: %+v", field.HclName, nestedModelName, err)
+			}
 		}
 
-		// first go pull out the nested item
-		nestedBlock, err := tb.getBlockValueForModel(field.HclName, nestedModel, dependencies, onlyRequiredFields)
-		if err != nil {
-			return nil, fmt.Errorf("retrieving block value for field %q containing nested List/Set reference to %q: %+v", field.HclName, nestedModelName, err)
+		// TODO support for lists of other basic types
+		if field.ObjectDefinition.NestedObject.Type == resourcemanager.TerraformSchemaFieldTypeString {
+			nestedBlock = hclwrite.NewBlock(field.HclName, []string{})
 		}
```

Comment:
tests for this / can we revert this for now? I think this also wants to be an Attribute rather than a Block, so we probably need to extend the `needsBlock` function to check the `NestedObject` type isn't a simple/basic type for a list?